analysers/analyser_osmosis_indoor.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights Noémie Lehuby 2022                                         ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from .Analyser_Osmosis import Analyser_Osmosis
+
+sql10 = """
+SELECT
+  id,
+  ST_AsText(way_locate(linestring))
+FROM
+  ways
+WHERE
+  NOT is_polygon
+  AND tags?'indoor'
+  AND tags->'indoor' in ('room', 'corridor', 'area', 'level')
+"""
+
+sql20 = """
+SELECT
+  id,
+  ST_AsText(geom)
+FROM
+  nodes
+WHERE
+  tags?'indoor'
+  AND tags->'indoor' in ('room', 'corridor', 'area', 'level')
+"""
+
+sql30 = """
+CREATE TEMP TABLE indoor_rooms_with_door AS
+(
+SELECT
+  ways.id
+FROM
+  ways
+JOIN way_nodes ON ways.id = way_nodes.way_id
+JOIN nodes ON nodes.id = way_nodes.node_id
+WHERE
+  ways.is_polygon
+  AND ways.tags?'indoor'
+  AND ways.tags->'indoor' = 'room'
+  AND nodes.tags?'door'
+)
+"""
+
+sql31 = """
+SELECT
+  ways.id,
+  ST_AsText(ST_Centroid(ways.linestring))
+FROM
+  ways
+LEFT JOIN indoor_rooms_with_door ON indoor_rooms_with_door.id = ways.id
```

Comment:
Here. Reuse indoor rooms table, and just join for the doors.

Reply:
ok, but I don't need to join on the doors, I need to set apart the rooms that have at least one node which is a door, so I think I still need the temp table for the rooms with doors :thinking: